### ISSUE TITLE
[3.2] Backwards incompatible cart loading from session

### ISSUE THREAD

**Issue by franticpsyx** [woocommerce/woocommerce#16568]
It's probably early as 3.2 beta hasn't arrived, but this was reported to me earlier so I thought I'd share:

In `WC_Cart_Session::get_cart_from_session` cart items are first loaded into a local `$cart_contents` array before they are copied into the cart using `WC_Cart::set_cart_contents`.

In WC 3.1, each item was copied into the `WC_Cart::cart_contents` array as it was loaded from session data, which means that any code hooked into the `woocommerce_get_cart_item_from_session` filter could "find" all items already loaded in `WC_Cart::cart_contents`.

TL;DR -- third-party code that hooks into `woocommerce_get_cart_item_from_session` and expects to find another item in the `WC_Cart::cart_contents` array will fail.

An obvious fix is to call `WC_Cart::set_cart_contents` in each iteration.

**Comment by mikejolley** [comment]
💩

**Comment by mikejolley** [comment]
See https://github.com/woocommerce/woocommerce/pull/16570
